MetaMask/metamask-extension
# feat: added check for if the selected account is BTC in transaction-list

## **Description**

Updates the activity tab view to display the message "Bitcoin activity is not supported" if the selected account is a Bitcoin account. It also adds a button to redirect to the Bitcoin block explorer.

[![Open in GitHub Codespaces](https://github.com/codespaces/badge.svg)](https://codespaces.new/MetaMask/metamask-extension/pull/25642?quickstart=1)

## **Related issues**

Fixes: https://github.com/MetaMask/accounts-planning/issues/499

## **Manual testing steps**

1. <details><summary>Create a Bitcoin account</summary>
   <br />

    1. Clone the [Bitcoin Manager](https://github.com/MetaMask/bitcoin) snap
    2. Change `emitEvents` [here](https://github.com/MetaMask/bitcoin/blob/main/packages/snap/src/rpcs/create-account.ts#L34) from false to **true**
    3. Run `yarn && yarn start`
    4. Go to [http://localhost:8000/](http://localhost:8000/)
    5. Create a Bitcoin account

</details>

6. Ensure that the new Bitcoin account is your selected account and navigate to the activity tab in the extension
7. You _should_ see the message "Bitcoin activity is not supported"
8. You _should_ see a button that says "View on block explorer" that opens to "https<area>://blockstream.info/testnet/address/_YourBitcoinAddress_"

## **Screenshots/Recordings**



### **Before**

<img width="359" alt="Screenshot 2024-07-02 at 7 23 43 PM" src="https://github.com/MetaMask/metamask-extension/assets/91970214/129d2e24-bc30-4759-94d9-bad76806d714">

### **After**

<img width="354" alt="Screenshot 2024-07-03 at 11 47 19 AM" src="https://github.com/MetaMask/metamask-extension/assets/91970214/62e50821-b802-49d7-9441-d5bed71f70b2">

## **Pre-merge author checklist**


## **Pre-merge reviewer checklist**
